app/src/main/java/de/jakobclass/transittracker/MapActivity.kt

```diff
@@ -19,28 +21,36 @@ import com.google.android.gms.maps.GoogleMap.OnCameraChangeListener
 import com.google.android.gms.maps.OnMapReadyCallback
 import com.google.android.gms.maps.SupportMapFragment
 import com.google.android.gms.maps.model.*
+import de.jakobclass.transittracker.animation.LatLngTypeEvaluator
 import de.jakobclass.transittracker.models.Stop
 import de.jakobclass.transittracker.factories.BitmapFactory
-import de.jakobclass.transittracker.services.StopService
-import de.jakobclass.transittracker.services.StopServiceDelegate
+import de.jakobclass.transittracker.models.Vehicle
+import de.jakobclass.transittracker.models.VehicleDelegate
+import de.jakobclass.transittracker.services.ApiService
+import de.jakobclass.transittracker.services.ApiServiceDelegate
 
-class MapActivity : FragmentActivity(), OnMapReadyCallback, ConnectionCallbacks, OnCameraChangeListener, StopServiceDelegate {
+class MapActivity : FragmentActivity(), OnMapReadyCallback, ConnectionCallbacks,
+        OnCameraChangeListener, ApiServiceDelegate, VehicleDelegate {
+
+    private val apiService: ApiService
+        get() = (application as Application).apiService
+    private var googleApiClient: GoogleApiClient? = null
     private var map: GoogleMap? = null
     private lateinit var mapFragment: SupportMapFragment
-    private var googleApiClient: GoogleApiClient? = null
-    private val stopService: StopService
-        get() = (application as Application).stopService
+    private val vehicleMarkers = mutableMapOf<Vehicle, Marker>()
 
     private val REQUEST_CODE_ACCESS_FINE_LOCATION = 1
     private val REQUIRED_LOCATION_PERMISSION = Manifest.permission.ACCESS_FINE_LOCATION
+    private val screenDensity: Float
+        get() = resources.displayMetrics.density
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         setContentView(R.layout.activity_map)
         mapFragment = supportFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
         mapFragment.getMapAsync(this)
-        stopService.delegate = this
+        apiService.delegate = this
```

Comment:
That's why I am using weak references, so that the user of the service doesn't have to care about that.